### PR TITLE
chore: update rn 72

### PR DESCRIPTION
**Pull request by ice-orion**
* Update React-Native to 0.72 + update several other libs (reanimated, gesture-handler, ...)
* Replace `react-native-vision-camera` with `expo-camera` for scanning QR codes (`react-native-vision-camera` was a blocker for update)
* Fix typescript issues
* Fix Welcome stack initial screen issue

### CODE REVIEW

File: android/app/src/release/java/io/ice/app/ReactNativeFlipper.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: ice License 1.0
+
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * <p>This source code is licensed under the MIT license found in the LICENSE file in the root
+ * directory of this source tree.
+ */
+package com.rndiffapp;
```
Comment (ice-atlas) [inline android/app/src/release/java/io/ice/app/ReactNativeFlipper.java]: check this pls

File: src/screens/Modals/CountrySelect/index.tsx [hunk 2]
```diff
@@ -40,7 +40,7 @@ export const CountrySelect = () => {
   );
 
   const renderItem = useCallback(
-    ({item}) => {
+    ({item}: {item: Country}) => {
```
Comment (ice-atlas) [inline src/screens/Modals/CountrySelect/index.tsx]: Let's use 
```
const renderItem: ListRenderItem<Country> = useCallback(...
```

File: android/app/src/release/java/io/ice/app/ReactNativeFlipper.java (hunk 1 above)
Comment (ice-orion) [reply android/app/src/release/java/io/ice/app/ReactNativeFlipper.java]: yea, already fixed

File: src/screens/Modals/CountrySelect/index.tsx (hunk 2 above)
Comment (ice-orion) [reply src/screens/Modals/CountrySelect/index.tsx]: done

**Review by ice-atlas — DISMISSED** [review]

**Review by ice-ares — APPROVED** [review]

**Review by ice-hades — APPROVED** [review]

**Review by ice-atlas — APPROVED** [review]

**Pull request merged**